lib/elastic_record/as_document.rb

```diff
@@ -16,27 +16,43 @@ def as_partial_update_document
 
       changed_fields.each_with_object({}) do |field, result|
         if field_mapping = mappings[field]
-          result[field] = elastic_search_value field, field_mapping
+          result[field] = value_for_elastic_search field, field_mapping
         end
       end
     end
 
-    def elastic_search_value(field, mapping)
+    def value_for_elastic_search(field, mapping)
       value = try field
       return if value.nil?
 
-      value = case mapping[:type]&.to_sym
-              when :object
-                value.respond_to?(:as_search_document) ? value.as_search_document : value
-              when :nested
-                value.map(&:as_search_document)
-              else
-                value
-              end
+      value =
+        case mapping[:type]&.to_sym
+        when :object
+          value_for_elastic_search_object(value)
+        when :nested
+          value.map { |entry| value_for_elastic_search_object(entry) }
+        when :integer_range, :float_range, :long_range, :double_range, :date_range
+          value_for_elastic_search_range(value)
+        else
+          value
+        end
 
       if value.present? || value == false
         value
       end
     end
+
+    def value_for_elastic_search_object(object)
+      object.respond_to?(:as_search_document) ? object.as_search_document : object
+    end
+
+    def value_for_elastic_search_range(range)
+      if range.begin <= range.end
```

Comment:
Delete this. Allow it to crash.

Reply:
The crash will never happen though because ORM layer will never create the scenario.

Reply:
elastic_record does not depend on any ORM layer besides active_model.  You are correct that superstore will prevent this, though.

Reply:
removed in 8fc5191ec2216707740d5bd5148bc0dd9e158080

tests reflect expected behavior